src/util/getDisplayLineMap.ts

```diff
@@ -3,20 +3,22 @@ import { concat, flatten, flow, range, uniq } from "lodash";
 
 /**
  * Returns a map from line numbers in the file to display lines, which skip
- * folded sections etc.  Note that if the cursor is currently offscreen, it
- * will add a line for the cursor before or after the screen lines depending
- * whether cursor is above or below last visible line.
+ * folded sections etc.
  *
  * @param editor A visible editor
+ * @param extraLines Optional extra lines to include in the map
  */
-export function getDisplayLineMap(editor: TextEditor) {
+export function getDisplayLineMap(
+  editor: TextEditor,
+  extraLines: number[] = [],
+): Map<number, number> {
   return new Map(
     flow(
       flatten,
       uniq,
     )(
       concat(
-        [[editor.selections[0].start.line]],
```

Comment:
...but `start` here 😳

This means that if `active` is not on the same line as `start`, and not visible, our token prioritisation breaks